play-json/jvm/src/main/scala/play/api/libs/json/jackson/JacksonJson.scala

```diff
@@ -96,10 +95,15 @@ private[jackson] class JsValueSerializer(jsonConfig: JsonConfig) extends JsonSer
         val stripped = stripTrailingZeros(v.bigDecimal)
         val raw      = if (shouldWritePlain) stripped.toPlainString else stripped.toString
 
-        if (raw.indexOf('E') < 0 && raw.indexOf('.') < 0)
-          json.writeTree(new BigIntegerNode(new BigInteger(raw)))
+        if (raw.exists(c => c == 'E' || c == '.'))
+          json.writeNumber(raw)
         else
-          json.writeTree(new DecimalNode(new JBigDec(raw)))
+          json match {
+            case _: TokenBuffer =>
+              json.writeTree(new BigIntegerNode(new BigInteger(raw)))
```

Comment:
I have a change in jackson 2.18 which means this line can be changed to something that avoids the BigInteger parsing.

Reply:
@pjfanning I merged the test (#1077) and also the main branch is on Jackson 2.19.1 now thanks to your pull request. Can you please now update this line with what you mentioned (as we are now on Jackson 2.18+)? thanks!